DecipherNow/gm-fabric-dashboard
# 1043 go functions crash

Resolves #1043 

## Proposed Changes

  - The "functions" and "Routes" keys must now match the first characters of the key to prevent errors from occuring if the keys contain the work "function" or "route" anywhere else in the key
  - Added supporting unit tests and fixes test regressions causes by the changes to the mock data needed to recreate the issue.
  
## How to Test

  - Look at the mock data and notice that it now contains a series of keys starting with `route/functionalroles/GET/`. Previously, this was causing the key to match as a function
  - Run `npm test` to ensure that the unit tests are running cleanly
  - Run `npm start` and `npm mock-sds` and navigate to the functions tab of a Go microservice to ensure that the app does not crash.
  - If desired, change the import in `sds.js` from `metricsWithFunctions` to `metricsWithoutFunctions` and restart mock-sds to ensure the app handles both scenarios.
